### PR TITLE
[SYCL] Don't expose vector of booleans as storage format

### PR DESCRIPTION
**Pull request by bader**
This reverts commit 1b8c0089ad79ee4765051ddfc722abc31d3ddac9.
Revert "[SYCL] Enable ext_vector_type of boolean type for SYCL language."

Replaces vector of booleans with vector of integers to store results of
logical/relational operations.

Similar change is propagated to SPIR-V built-ins declarations.

Signed-off-by: Mariya Podchishchaeva <mariya.podchishchaeva@intel.com>
Signed-off-by: Alexey Bader <alexey.bader@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/boolean.hpp [hunk 1]
```diff
@@ -56,11 +56,11 @@ template <int N> struct Boolean {
   using DataType = element_type[N];
 #endif
 
-  Boolean() : value{false} {}
+  Boolean() : value{0} {}
 
   Boolean(std::initializer_list<element_type> l) {
     for (size_t I = 0; I < N; ++I) {
-      value[I] = *(l.begin() + I);
+      value[I] = (!!*(l.begin() + I)) * (-1);
```
Comment (Fznamznon) [inline sycl/include/CL/sycl/detail/boolean.hpp]: It's hard to understand what is happening here (probably I still don't, so suggestion is not correct). Can we rewrite it to make it dummy?
```suggestion
      value[I] = (*(l.begin() + I))? -1 : 0;
```

File: clang/lib/Sema/SPIRVBuiltins.td [hunk 2]
```diff
@@ -816,13 +816,36 @@ foreach name = ["Dot"] in {
 }
 
 foreach name = ["Any", "All"] in {
-  def : SPVBuiltin<name, [Bool, GenTypeBoolVecAndScalar], Attr.Const>;
+  def : SPVBuiltin<name, [Bool, Bool], Attr.Const>;
+}
+
+foreach name = ["Any", "All"] in {
+  def : SPVBuiltin<name, [Bool, GenTypeSCharVecNoScalar], Attr.Const>;
+}
```
Comment (Naghasan) [inline clang/lib/Sema/SPIRVBuiltins.td]:
```suggestion
  def : SPVBuiltin<name, [Bool, Bool], Attr.Const>;
  def : SPVBuiltin<name, [Bool, GenTypeSCharVecNoScalar], Attr.Const>;
}
```

File: clang/lib/Sema/SPIRVBuiltins.td [hunk 3]
```diff
@@ -816,13 +816,36 @@ foreach name = ["Dot"] in {
...
+                "Ordered", "Unordered",
+                "FOrdEqual", "FUnordEqual",
+                "FOrdNotEqual", "FUnordNotEqual",
+                "FOrdLessThan", "FUnordLessThan",
+                "FOrdGreaterThan", "FUnordGreaterThan",
+                "FOrdLessThanEqual", "FUnordLessThanEqual",
+                "FOrdGreaterThanEqual", "FUnordGreaterThanEqual"] in {
+  def : SPVBuiltin<name, [Bool, Float, Float], Attr.Const>;
+  def : SPVBuiltin<name, [Bool, Double, Double], Attr.Const>;
+  def : SPVBuiltin<name, [Bool, Half, Half], Attr.Const>;
```
Comment (Naghasan) [inline clang/lib/Sema/SPIRVBuiltins.td]: same as above, the definitions can put into the same loop

File: clang/lib/Sema/SPIRVBuiltins.td [hunk 4]
```diff
@@ -367,7 +367,7 @@ def IntLongFloatGenType1   : GenericType<"IntLongFloatGenType1", TLIntLongFloats
 
 // GenType definitions for every single base type (e.g. fp32 only).
 // Names are like: GenTypeFloatVecAndScalar.
-foreach Type = [Bool, Char, UChar, Short, UShort,
+foreach Type = [Bool, Char, UChar, SChar, Short, UShort,
```
Comment (Naghasan) [inline clang/lib/Sema/SPIRVBuiltins.td]: `Bool` should be removed from here

Comment (bader) [reply clang/lib/Sema/SPIRVBuiltins.td]: Updated. Let me know if I missed anything.

**Pull request merged**